### PR TITLE
Replace specials helpers arguments by dedicated functions

### PR DESCRIPTION
**Pull request by lyrixx**
fixes https://github.com/jolicode/castor/pull/108

TODO:

* [x] Update code
* [x] update example / tests
* [x] Update doc

---

**Important**

* this is **not possible** to name a function `var()` (reserved keyword)
* it's a bad idea to name function application, command, context, because it will conflict with associated types!

### CODE REVIEW

**Comment by lyrixx** [comment]
I think I forgot the readme.

File: doc/05-context.md [hunk 1]
```diff
@@ -18,12 +18,17 @@ with `Castor\Context`:
 ```php
 use Castor\Context;
 
+use function Castor\get_context;
+
 #[AsTask]
```
Comment (pyrech) [inline doc/05-context.md:23]: First sentence of the chapter need to be updated too

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## 0.4.0 (not released yet)
 
+* [BC Break] specials helpers required via function arguments are now accessible via functions
```
Comment (pyrech) [inline CHANGELOG.md:5]:
```suggestion
* [BC Break] replace specials helpers arguments by dedicated functions
```

**Review by pyrech — COMMENTED** [review]
Should we add documentation for all the new functions for better discoverability?

File: doc/05-context.md (hunk 1 above)
Comment (lyrixx) [reply doc/05-context.md:23]: Thanks

**Comment by lyrixx** [comment]
>Should we add documentation for all the new functions for better discoverability?

Good idea! I reviewed all the doc BTW! I hope you'll like the new version

File: doc/05-context.md [hunk 3]
```diff
@@ -12,18 +12,46 @@ context is created.
...
+```php
+use Castor\Attribute\AsTask;
+
+use function Castor\get_context;
+use function Castor\variable;
+
+#[AsTask]
+function foo(): void
+{
+    $foobar = variables('foobar', 'default value');
```
Comment (pyrech) [inline doc/05-context.md:46]:
```suggestion
    $foobar = variable('foobar', 'default value');
```

File: doc/06-helper.md [hunk 4]
```diff
@@ -1,73 +1,82 @@
...
 You can check
 the [Symfony documentation](https://symfony.com/doc/current/console/style.html)
 for more information about this class and how to use it.
 
-# Filesystem
+## Filesystem
 
-The `Symfony\Component\Filesystem\Filesystem` class provides OS-independent
-utilities for filesystem operations and for file/directory paths manipulation.
+The `fs()`: returns an object that provides OS-independent utilities for
```
Comment (pyrech) [inline doc/06-helper.md:54]:
```suggestion
The `fs()` function returns an object that provides OS-independent utilities for
```

File: doc/06-helper.md [hunk 5]
```diff
@@ -77,24 +86,46 @@ You can check
...
+* `get_input()` returns the current
+  [`Input`](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Console/Output/OutputInterface.php)
+* `get_output()` returns the current
+  [`Output`](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Console/Input/InputInterface.php)
+
+## Other helpers
+
+* `get_context()` returns the initial `Context`. See the [context
+  documentation](./05-context.md) for mor information
+* `variables()` returns a variables stored in the  `Context`. See the [context
```
Comment (pyrech) [inline doc/06-helper.md:128]:
```suggestion
* `variable()` returns a variable stored in the  `Context`. See the [context
```

**Comment by lyrixx** [comment]
@pyrech Thanks for the review 👍🏼

**Review by pyrech — APPROVED** [review]

**Pull request merged**